lib/helpers.js

```diff
@@ -127,11 +130,58 @@ const parseResults = (editor, results, filePath, showIgnored) => {
   return toReturn;
 };
 
+function isDirectory(dirPath) {
+  try {
+    return fs.statSync(dirPath).isDirectory();
+  } catch (e) {
+    return false;
+  }
+}
+
+export function findStylelintDirectory(modulesDir) {
+  const stylelintDir = path.join(modulesDir || '', 'stylelint');
+  if (isDirectory(stylelintDir)) {
+    return stylelintDir;
+  }
+  return stylelintLocalPath;
+}
+
+export function getStylelintFromDirectory(modulesDir) {
+  const stylelintDir = findStylelintDirectory(modulesDir);
+  try {
+    // eslint-disable-next-line import/no-dynamic-require
+    return require(stylelintDir);
+  } catch (e) {
+    if (e.code === 'MODULE_NOT_FOUND') {
+      throw new Error('Stylelint not found, try restarting Atom to clear caches.');
+    }
+    // eslint-disable-next-line import/no-dynamic-require
+    return require(stylelintLocalPath);
+  }
+}
+
+export function refreshModulesPath(modulesDir) {
+  if (lastModulesPath !== modulesDir) {
+    lastModulesPath = modulesDir;
+    process.env.NODE_PATH = modulesDir || '';
+    // eslint-disable-next-line no-underscore-dangle
+    require('module').Module._initPaths();
```

Comment:
`linter-eslint` (which I'm guessing you are copying) [gets away with this](https://github.com/AtomLinter/linter-eslint/blob/7f791653480ce30c1ffe9ab99c6c84e34fe7e7fe/src/worker-helpers.js#L102-L109) because it is running that code in a worker thread completely separate from Atom, so this doesn't screw up _everything_ else there. Currently this isn't being done here though, so this is a huge no-no.

Maybe I need to resume working on the [tasks](https://github.com/AtomLinter/linter-stylelint/tree/tasks) branch...

Reply:
For now, cache the original path and restore it when done running `stylelint`.

Reply:
Something like this? c05c296

Reply:
Yea, that's as good as we can get until this is moved into a `Task` of its own. At least this way the window of potential breakage is only a few dozen `ms`.